### ISSUE TITLE
on linux register on another poll instance works after deregister

### ISSUE THREAD

**Issue by dtacalau** [tokio-rs/mio#1203]
According to deregister docs:
```
    /// A handle can be passed back to `register` after it has been
    /// deregistered; however, it must be passed back to the **same** `Poll`
    /// instance.
```

I noticed this issue when testing #1201. Linux does not follow doc constraints for deregister. The following testfail only on `Test --release Linux`:

running 15 tests
test deregister_without_register ... ok
test double_register_different_token ... ok
test drop_cancels_interest_and_shuts_down ... ok
test is_send_and_sync ... ok
test poll_closes_fd ... ok
test poll_erroneous_registration ... ok
test poll_registration ... ok
test register_deregister_multiple_poll_instances ... FAILED
test register_reregister_token_usage ... ok
test registry_behind_arc ... ok
test reregister_overrides_interest ... ok
test add_then_drop ... ok
test reregister_without_register ... ok
test zero_duration_polls_events ... ok
test run_once_with_nothing ... ok

failures:

---- register_deregister_multiple_poll_instances stdout ----
thread 'register_deregister_multiple_poll_instances' panicked at 'called `Result::unwrap_err()` on an `Ok` value: ()', src/libcore/result.rs:1165:5
stack backtrace:
   0:     0x55e9fd8d9194 - backtrace::backtrace::libunwind::trace::hda41dbcdfba36aa0
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1:     0x55e9fd8d9194 - backtrace::backtrace::trace_unsynchronized::h1a8d6e1f8cb3f5d4
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2:     0x55e9fd8d9194 - std::sys_common::backtrace::_print_fmt::h610c4127487e10da
                               at src/libstd/sys_common/backtrace.rs:76
   3:     0x55e9fd8d9194 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h0722dc552e01bd1d
                               at src/libstd/sys_common/backtrace.rs:60
   4:     0x55e9fd8f558c - core::fmt::write::h01edf6dd68a42c9c
                               at src/libcore/fmt/mod.rs:1030
   5:     0x55e9fd8903e5 - std::io::Write::write_fmt::hd48b46e5c13c6612
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/io/mod.rs:1412
   6:     0x55e9fd8d3201 - std::io::impls::<impl std::io::Write for alloc::boxed::Box<W>>::write_fmt::hc4feb234d1aa0439
                               at src/libstd/io/impls.rs:141
   7:     0x55e9fd8db625 - std::sys_common::backtrace::_print::hd8d5d08a1795e743
                               at src/libstd/sys_common/backtrace.rs:64
   8:     0x55e9fd8db625 - std::sys_common::backtrace::print::hf89a79e3921a2366
                               at src/libstd/sys_common/backtrace.rs:49
   9:     0x55e9fd8db625 - std::panicking::default_hook::{{closure}}::h3a8f42beb3bb8ae3
                               at src/libstd/panicking.rs:196
  10:     0x55e9fd8db2ac - std::panicking::default_hook::h8f803b0bc31a5c37
                               at src/libstd/panicking.rs:207
  11:     0x55e9fd8dbd25 - std::panicking::rust_panic_with_hook::h825f041245da8739
                               at src/libstd/panicking.rs:473
  12:     0x55e9fd8db8c2 - std::panicking::continue_panic_fmt::hbe0378e33481e81b
                               at src/libstd/panicking.rs:380
  13:     0x55e9fd8db7b6 - rust_begin_unwind
                               at src/libstd/panicking.rs:307
  14:     0x55e9fd8f2a6a - core::panicking::panic_fmt::h527855ce0bc891f6
                               at src/libcore/panicking.rs:85
  15:     0x55e9fd8f2b67 - core::result::unwrap_failed::ha8b77e6004f0ba38
                               at src/libcore/result.rs:1165
  16:     0x55e9fd87e789 - core::ops::function::FnOnce::call_once::h9fd674a20cc423d3
  17:     0x55e9fd89a2df - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h60313ef8cd80fa37
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
  18:     0x55e9fd8df74a - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:80
  19:     0x55e9fd8b4aee - std::panicking::try::h218111062acbd8af
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:271
  20:     0x55e9fd8b4aee - std::panic::catch_unwind::hb43bd7e3eefd561b
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:394
  21:     0x55e9fd8b4aee - test::run_test::run_test_inner::{{closure}}::h61999fd5b32b527c
                               at src/libtest/lib.rs:1413
  22:     0x55e9fd88fb55 - std::sys_common::backtrace::__rust_begin_short_backtrace::h94f0165e3063a572
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/sys_common/backtrace.rs:126
  23:     0x55e9fd893d55 - std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::ha13ef4f272e8d7f5
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/mod.rs:470
  24:     0x55e9fd893d55 - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h4be9743fa104fe58
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:315
  25:     0x55e9fd893d55 - std::panicking::try::do_call::hc01172ae30fe685a
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:292
  26:     0x55e9fd8df74a - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:80
  27:     0x55e9fd8942c2 - std::panicking::try::h28c280234b112988
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:271
  28:     0x55e9fd8942c2 - std::panic::catch_unwind::h80abb9ed16b7a343
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:394
  29:     0x55e9fd8942c2 - std::thread::Builder::spawn_unchecked::{{closure}}::h079ad7fe053e2851
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/mod.rs:469
  30:     0x55e9fd8942c2 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h66965f502b707a1a
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libcore/ops/function.rs:227
  31:     0x55e9fd8cf96f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h483711add4ba2330
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
  32:     0x55e9fd8dee10 - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h7605b45eb29ed0be
                               at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:922
  33:     0x55e9fd8dee10 - std::sys_common::thread::start_thread::h557b0c2dc4449562
                               at src/libstd/sys_common/thread.rs:13
  34:     0x55e9fd8dee10 - std::sys::unix::thread::Thread::new::thread_start::h7c2a7f9b68fe4bba
                               at src/libstd/sys/unix/thread.rs:79
  35:     0x7fee54eb46ba - start_thread
  36:     0x7fee549d241d - clone
  37:                0x0 - <unknown>

failures:
    register_deregister_multiple_poll_instances

test result: FAILED. 14 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--test poll'
##[error]Bash exited with code '101'.
##[section]Finishing: cargo test --release

**Comment by Thomasdezeeuw** [comment]
This is more or less the same answer as I commented here https://github.com/tokio-rs/mio/issues/1202#issuecomment-563236612. For cross-platform application this is true, but some platforms are more lenient. I don't consider this a problem.

**Comment by dtacalau** [comment]
>  I don't consider this a problem really.

@Thomasdezeeuw  I'm fine with just documenting this. WDYT?

**Comment by Thomasdezeeuw** [comment]
Same answer as here: https://github.com/tokio-rs/mio/issues/1202#issuecomment-563297093.

**Comment by dtacalau** [comment]
Closed via #1208
